base/src/main/scala/co/uproot/abandon/Utils.scala

```diff
@@ -12,4 +13,62 @@ object FileUtils {
     return new File(dirName).listFiles.filter({d => d.isDirectory && d.getCanonicalPath.matches(regex)}).sorted
   }
 
+  /*
+   * In theory, this can blow up your stack, but in practice, it won't.
+   * To do that, there would have to be very deep directory structure.
+   */
+  private def listDirTree(path: String): List[File] = {
```

Comment:
It's not tail recursive, because there is "flatten" outside, AFAIK,
I checked nio: there is visitor pattern with directory walker, but I found it easier to list all files and filter whole set then. Current implementation can handle all files on Unix filesystem without problem. And code uses java.nio.file.PathMatcher so that part is from nio.

So maybe it's fair trade-off?